manager/state/raft/raft.go

```diff
@@ -1302,6 +1303,46 @@ func (n *Node) reportNewAddress(ctx context.Context, id uint64) error {
 	return n.transport.UpdatePeerAddr(id, newAddr)
 }
 
+// RaftMessageStream is the server endpoint for streaming Raft messages.
+// It accepts a stream of raft messages to be processed on this raft member,
+// returning a ProcessRaftMessageResponse when processing of the streamed messages is complete.
+// It is called from the Raft leader, which uses it to stream messages to this raft member.
+func (n *Node) RaftMessageStream(stream api.Raft_RaftMessageStreamServer) error {
+	var recvdMsg, assembledMessage *api.ProcessRaftMessageRequest
+	var err error
+	for {
+		recvdMsg, err = stream.Recv()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			log.G(context.Background()).WithError(err).Error("error while reading from stream")
+			return err
+		}
+
+		if assembledMessage == nil {
+			assembledMessage = recvdMsg
+			continue
+		}
+
+		// Append the received snapshot data.
+		if recvdMsg.Message.Type == raftpb.MsgSnap {
+			assembledMessage.Message.Snapshot.Data = append(assembledMessage.Message.Snapshot.Data, recvdMsg.Message.Snapshot.Data...)
+		} else {
+			log.G(context.Background()).Errorf("Ignoring unexpected message type received on stream: %d", recvdMsg.Message.Type)
+		}
+	}
+
+	// We should have the complete snapshot. Verify and process.
+	if err == io.EOF {
+		_, err = n.ProcessRaftMessage(context.Background(), assembledMessage)
```

Comment:
Unsure if I think we should use the stream context here or not.

Reply:
maybe it's better not to discard the first return value, and instead use it as the return value for `stream.SendAndClose` below?

Reply:
>Unsure if I think we should use the stream context here or not.

Yea, I'm unclear on this also. I'll think more about this.